### PR TITLE
:sparkles: Upgrade kubebuilder dep from 3.3.0 to 3.4.1

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Upgrade kubebuilder dep from 3.3.0 to 3.4.1

### CODE REVIEW

File: testdata/go/v2/memcached-operator/Makefile [hunk 1]
```diff
@@ -145,37 +145,31 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
...
-KUSTOMIZE=$(shell which kustomize)
-endif
+kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
+$(KUSTOMIZE): $(LOCALBIN)
+	curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN)
+
+.PHONY: controller-gen
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+$(CONTROLLER_GEN): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
```
Comment (camilamacedo86) [inline testdata/go/v2/memcached-operator/Makefile:172]: @ryantking we upgrade the go/v2, and I am looking we cannot upgrade this one. Why?
Go/v2 only works with crd api version v1beta1, which means that this layout cannot be upgraded to use the latest versions of controller runtime and controller tools. See that we update the controller tools here.

Could you please do a test locally? Looking at the code, we cannot. see https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v2/api.go#L94-L95)

So, we need to:
- Revert the makefile changes there. 

c/c @rashmigottipati

File: testdata/go/v2/memcached-operator/Dockerfile [hunk 2]
```diff
@@ -15,7 +15,7 @@ COPY api/ api/
 COPY controllers/ controllers/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
```
Comment (everettraven) [inline testdata/go/v2/memcached-operator/Dockerfile:18]: I noticed that there was a [commit](https://github.com/kubernetes-sigs/kubebuilder/commit/c9f63e8147a956547d7574f9abea0a3569d720df) that added the `GO111MODULE=on` back to the Dockerfile scaffold template for go/v2. Will this difference cause any issues for users wanting to scaffold with go/v2?

**Review by everettraven — APPROVED** [review]
Looks good to me. I think it would be a good idea to get another set of eyes on this before merging.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**